### PR TITLE
[VENTUS][fix] Fix insert vmv instruction bug when vmv instruction is in JOIN MBB

### PR DESCRIPTION
**Pull request by zhoujingya**

### CODE REVIEW

File: llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp [hunk 1]
```diff
@@ -231,12 +231,13 @@ bool VentusInsertJoinToVBranch::checkJoinMBB(MachineBasicBlock &MBB) const {
     if (MI.getOpcode() != RISCV::VMV_V_X)
       continue;
 
-    // To be removed vmv.v instruction flag
-    bool NeedToBeErased = false;
+    // The number of MBB needed to be inserted VMV instruction
+    unsigned NeedToBeInsertMBBNum = 0;
```
Comment (dodohack) [inline llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp:235]:
```
    // The number of MBBs where VMV instruction should be inserted into.
    unsigned NumMBB = 0;
```

File: llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp [hunk 2]
```diff
@@ -257,26 +258,25 @@ bool VentusInsertJoinToVBranch::checkJoinMBB(MachineBasicBlock &MBB) const {
 
       // Check if define instruction is in the predecessors or not
       for (auto *Pre : MBB.predecessors()) {
-        bool NeedToBeInsert = false;
         MachineBasicBlock::iterator Insert = Pre->begin();
```
Comment (dodohack) [inline llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp:261]: `Insert` -> `I`

Comment (dodohack) [inline llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp:261]: Iterator naming convention: `Insert` -> `I`.

File: llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp [hunk 3]
```diff
@@ -231,12 +231,13 @@ bool VentusInsertJoinToVBranch::checkJoinMBB(MachineBasicBlock &MBB) const {
...
-    // To be removed vmv.v instruction flag
-    bool NeedToBeErased = false;
+    // The number of MBB needed to be inserted VMV instruction
+    unsigned NeedToBeInsertMBBNum = 0;
     assert(MI.getOperand(1).isReg() && "unexpected operator");
     auto Defines = MR.def_instructions(MI.getOperand(1).getReg());
     bool IsInSameBlock = false;
-
+    SmallVector<std::pair<MachineBasicBlock *, MachineBasicBlock::iterator>>
+        MBBMaybeInsertedInstr;
```
Comment (dodohack) [inline llvm/lib/Target/RISCV/VentusInsertJoinToVBranch.cpp:240]: Can we have a better name for `MBBMaybeInsertedInstr`?

**Review by dodohack — CHANGES_REQUESTED** [review]
Coding style fix please.

**Review by wangqinfan — APPROVED** [review]
LGTM

**Pull request closed without merging**